### PR TITLE
[DependencyInjection][FrameworkBundle] Leverage array_is_list

### PR DESCRIPTION
**Pull request by jderusse**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

Use `array_is_list` instead of `count`+`range`+`array_key`

take over #44156

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/composer.json [hunk 1]
```diff
@@ -28,6 +28,7 @@
         "symfony/http-kernel": "^5.4|^6.0",
         "symfony/polyfill-mbstring": "~1.0",
         "symfony/polyfill-php80": "^1.16",
+        "symfony/polyfill-php81": "^1.16",
```
Comment (fancyweb) [inline src/Symfony/Bundle/FrameworkBundle/composer.json]:
```suggestion
        "symfony/polyfill-php81": "^1.22",
```

The first release is 1.22.0.

Comment (derrabus) [inline src/Symfony/Bundle/FrameworkBundle/composer.json]:
```suggestion
        "symfony/polyfill-php81": "^1.22",
```

The function that we need was introduced with 1.22.0. See symfony/polyfill#324

File: src/Symfony/Component/DependencyInjection/composer.json [hunk 2]
```diff
@@ -20,6 +20,7 @@
         "psr/container": "^1.1.1",
         "symfony/deprecation-contracts": "^2.1",
         "symfony/polyfill-php80": "^1.16",
+        "symfony/polyfill-php81": "^1.16",
```
Comment (derrabus) [inline src/Symfony/Component/DependencyInjection/composer.json]:
```suggestion
        "symfony/polyfill-php81": "^1.22",
```

**Review by fancyweb — COMMENTED** [review]
Looks like there is one more here: https://github.com/symfony/symfony/blob/5.4/src/Symfony/Component/Intl/Data/Bundle/Writer/TextBundleWriter.php#L84

**Review by derrabus — CHANGES_REQUESTED** [review]

**Comment by derrabus** [comment]
Looks like @fancyweb was a few seconds faster than me. 🙈

**Comment by jderusse** [comment]
:facepalm:

**Review by derrabus — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @jderusse.

**Pull request merged**